mne/event.py

```diff
@@ -827,3 +828,289 @@ def concatenate_events(events, first_samps, last_samps):
         events_out = np.concatenate((events_out, e2), axis=0)
 
     return events_out
+
+
+class Elekta_event(object):
+    """ Represents a trigger event in Elekta/Neuromag system.  """
+    # original DACQ variable names
+    vars = ['Name', 'Channel', 'NewBits', 'OldBits', 'NewMask',
+            'OldMask', 'Delay', 'Comment']
+
+    def __init__(self, name, channel, newbits, oldbits, newmask,
+                 oldmask, delay, comment):
+        self.name = name  # short name for the event
+        self.channel = channel  # stimulus channel, e.g. STI101
+        self.newbits = int(newbits)  # state after trigger transition
+        self.oldbits = int(oldbits)  # state before trigger transition
+        self.newmask = int(newmask)  # bitmask for post-transition state
+        self.oldmask = int(oldmask)  # bitmask for pre-transition state
+        self.delay = float(delay)  # delay to stimulus (s)
+        self.comment = comment  # verbose comment for the event
+        # non-DACQ vars
+        self.index = None  # index of event in DACQ list
+        self.in_use = False  # whether event is referred to by a category
+
+    def __repr__(self):
+        s = '<Elekta_event | '
+        s += 'index: %d ' % self.index
+        s += 'name: "%s", ' % self.name
+        s += 'comment: "%s", ' % self.comment
+        s += 'pre-state: %d, ' % self.oldbits
+        s += 'pre-mask: %d, ' % self.oldmask
+        s += 'post-state: %d, ' % self.newbits
+        s += 'post-mask: %d, ' % self.newmask
+        s += 'delay: %.3f (s)>' % self.delay
+        return s
+
+
+class Elekta_category(object):
```

Comment:
Ok, here's the current main API. Maybe it should be in a more visible place for discussion, not sure where though.

**Get info about events and averaging categories**
`eav = ElektaAverager(raw)`
`# prettyprint all categories and events in use`
`eav.categories`
`eav.events`

**Get a single category**
`vis = eav['visual_right']   # returns ElektaCategory object`

**Get corresponding epochs from a given Raw object**
`epochs = vis.epochs(some_raw)  # returns mne.Epochs object`

**Iterate over categories and compute corresponding averages**
`for cat in eav:`
&nbsp;&nbsp;&nbsp;&nbsp;`epochs = cat.epochs(raw)`
&nbsp;&nbsp;&nbsp;&nbsp;`evoked = epochs.average()`

**Get rejection info from averaging parameters and average accordingly**
`rej = eav.get_mne_rejection_dict()`
`evoked = eav['visual_left'].epochs(reject=rej)`